File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"

namespace raindrops {

std::string convert(int num) {
    std::string sounds;

    if (num % 3 == 0) {
        sounds += "Pling";
    }

    if (num % 5 == 0) {
        sounds += "Plang";
    }

    if (num % 7 == 0) {
        sounds += "Plong";
    }

    if (sounds.empty()) {
        sounds = std::to_string(num);
    }

    return sounds;
}
} // namespace raindrops

File: lua/raindrops/raindrops.lua
return function(n)
	local sounds = ""

	if n % 3 == 0 then
		sounds = sounds .. "Pling"
	end

	if n % 5 == 0 then
		sounds = sounds .. "Plang"
	end

	if n % 7 == 0 then
		sounds = sounds .. "Plong"
	end

	if sounds == "" then
		sounds = tostring(n)
	end

	return sounds
end
